### PR TITLE
Issue 718 test user.SigninUser

### PR DESCRIPTION
**Pull request by rjourde**

### CODE REVIEW

**Comment by santiaago** [comment]
* prefer a contains error test rather than a == err string
`!strings.Contains(gonawintest.ErrorString(err), test.err) `
https://github.com/taironas/gonawin/blob/ddf0b6265c8741513cd64ebf45c5af71122c5a3b/models/team_test.go#L109

* err string could be `"no valid query name."` rather than `"models/user: no valid query name."`, less verbose.
* `SigninUser` can also break with ` Unable to create user` have a think and see if it is posible to test this.
https://github.com/taironas/gonawin/blob/ee1a01dbe40b40846086b2a0024aa6c5f50766ba/models/user.go#L295
* we are currenlty ignoring the FindUser error
https://github.com/taironas/gonawin/blob/ee1a01dbe40b40846086b2a0024aa6c5f50766ba/models/user.go#L286
we should probably return it and also test that case.

Have a think at the scenarios and see have can be done :+1:

**Comment by rjourde** [comment]
* no scenario to get `unable to crrate user` error. And there is no error tes case in TestCreateUser.
Currently 'FindUser' is returning a nil User if no user has been found or if an error occurred. We could returned user = nil and error = nil when no user has been found. And we could returned user = nil and the error when an error occured. But then it will not be possible to test the error in TestSigninUser.

**Comment by santiaago** [comment]
ok thanks,

Some comments:

* I meant that the test should test `no valid query name` but the error returned by `Signin` still could be `models/user: no valid query name.`
* error should not have `.` at the end.

Things I don't like:
* `FindUser ` does not return an error
* `user.Publish` does not return an error

but I guess that will be done later.

**Comment by rjourde** [comment]
Yes we will do a pass for error handling later on.

File: models/user.go [hunk 1]
```diff
@@ -292,7 +290,7 @@ func SigninUser(w http.ResponseWriter, r *http.Request, queryName string, email
 		alias := ""
 		if userCreate, err := CreateUser(c, email, username, name, alias, isAdmin, GenerateAuthKey()); err != nil {
 			log.Errorf(c, "Signup: %v", err)
-			return nil, errors.New("models/user: Unable to create user.")
```
Comment (santiaago) [inline models/user.go]: I asked you to rollback this,

Comment (rjourde) [inline models/user.go]: You want to keep: `models/user: <error_message>`?

**Comment by santiaago** [comment]
yes on the `error` not on the `test`

**Pull request merged**